src/components/avatar/index.tsx

```diff
@@ -335,92 +263,136 @@ class Avatar extends PureComponent<AvatarProps> {
+          {...imageProps}
+        />
+      );
+    }
+  };
 
-    return (
-      <Container
-        style={[this.getContainerStyle(), containerStyle]}
-        ref={forwardedRef}
-        testID={testID}
-        onPress={onPress}
-        accessible={!_.isUndefined(onPress)}
-        accessibilityLabel={'Avatar'}
-        accessibilityRole={onPress ? 'button' : 'image'}
-        {...extractAccessibilityProps(this.props)}
-      >
-        <View
-          testID={`${testID}.container`}
-          style={[
-            this.getInitialsContainer(),
-            {backgroundColor: this.backgroundColor},
-            hasImage && this.styles.initialsContainerWithInset
-          ]}
-        >
-          {!_.isUndefined(text) && (
-            <Text numberOfLines={1} style={[{fontSize}, this.styles.initials, {color}]} testID={`${testID}.label`}>
-              {text}
-            </Text>
-          )}
+  const badge = () => {
+    if (!_.isEmpty(badgeProps)) {
+      return (
+        <Badge testID={`${testID}.onlineBadge`} {...badgeProps} size={badgeSize} containerStyle={_badgePosition}/>
+      );
+    }
+  };
+
+  const ribbon = () => {
+    if (ribbonLabel) {
+      return (
+        <View style={[_ribbonStyle, styles.ribbon, ribbonStyle]}>
+          <Text numberOfLines={1} text100 $textDefaultLight style={ribbonLabelStyle}>
+            {ribbonLabel}
+          </Text>
         </View>
-        {this.renderImage()}
-        {this.renderBadge()}
-        {hasCustomRibbon && this.renderCustomRibbon()}
-        {hasRibbonLabel && this.renderRibbon()}
-        {children}
-      </Container>
-    );
-  }
-}
+      );
+    }
+  };
 
-function createStyles(props: AvatarProps) {
-  const {labelColor} = props;
-  const styles = StyleSheet.create({
-    initialsContainerWithInset: {
-      top: 1,
-      right: 1,
-      bottom: 1,
-      left: 1
-    },
-    initials: {
-      color: labelColor,
-      backgroundColor: 'transparent',
-      lineHeight: undefined
-    },
-    ribbon: {
-      backgroundColor: Colors.$backgroundPrimaryHeavy,
-      paddingHorizontal: 6,
-      paddingVertical: 3
+  const _customRibbon = () => {
+    if (customRibbon) {
+      return <View style={_ribbonStyle}>{customRibbon}</View>;
     }
-  });
+  };
 
-  return styles;
-}
+  const Container = onPress ? TouchableOpacity : View;
+
+  return (
+    <Container
+      style={_containerStyle}
+      ref={ref}
+      testID={testID}
+      onPress={onPress}
+      accessible={!_.isUndefined(onPress)}
+      accessibilityLabel={'Avatar'}
+      accessibilityRole={onPress ? 'button' : 'image'}
+      {...accessibilityProps}
+    >
+      <View testID={`${testID}.container`} style={textContainerStyle}>
+        {!_.isUndefined(text) && (
+          <Text numberOfLines={1} style={textStyle} testID={`${testID}.label`}>
+            {text}
+          </Text>
+        )}
+      </View>
+      {image()}
+      {badge()}
+      {_customRibbon()}
+      {ribbon()}
```

Comment:
IMO, `renderImage`, `renderBadge`, `renderCustomRibbon`, and `renderRibbon` are better names when we talk about callable functions and not just memoized values.. WDYT?